src/comm.jl

```diff
@@ -1,264 +1,1531 @@
+               src = view(domain.commDataSend, idx:(idx+(xferFields * sendCount)))
+            else
+               error("unimplemented")
+            end
+
+            otherRank = myRank + domain.tp
+            req = MPI.Isend(src, otherRank, msgType + myRank, domain.comm)
+            domain.sendRequest[pmsg+1] = req
+            pmsg += 1
+         end
+
+         if rowMax && doSend
+            # contiguous memory
+            if packRow
+               srcOffset = dx*(dy - 1)
+               offset = pmsg * maxPlaneComm
+               for field in fields
+                  for i in 0:(dz-1)
+                     copyto!(domain.commDataSend, offset+i*dx , field, srcOffset+i*dx*dy, dx)
+                  end
+                  offset += sendCount
+               end
+               idx = pmsg * maxPlaneComm + 1
+               src = view(domain.commDataSend, idx:(idx+(xferFields * sendCount)))
+            else
+               error("unimplemented")
+            end
+
+            otherRank = myRank + domain.tp
+            req = MPI.Isend(src, otherRank, msgType + myRank, domain.comm)
+            domain.sendRequest[pmsg+1] = req
+            pmsg += 1
+         end
+
+         if colMin | colMax
+            # ASSUMING ONE DOMAIN PER RANK, CONSTANT BLOCK SIZE HERE
+            # static MPI_Datatype msgTypeCol ;
+            # static bool packCol ;
+            sendCount = dy * dz
+
+            # if (msgTypeCol == 0) {
+            #    /* Create an MPI_struct for field data */
+            #    if (ALLOW_UNPACKED_COL && packable) {
+
+            #       static MPI_Datatype msgTypePoint ;
+            #       static MPI_Datatype msgTypePencil ;
+
+            #       /* dy points per pencil */
+            #       MPI_Type_vector(dy, 1, dx, baseType, &msgTypePoint) ;
+            #       MPI_Type_commit(&msgTypePoint) ;
+
+            #       /* dz pencils per plane */
+            #       MPI_Type_vector(dz, 1, dx*dy, msgTypePoint, &msgTypePencil) ;
+            #       MPI_Type_commit(&msgTypePencil) ;
+
+            #       MPI_Type_vector(xferFields, 1, (fieldData[1] - fieldData[0]),
+            #                       msgTypePencil, &msgTypeCol) ;
+            #       MPI_Type_commit(&msgTypeCol) ;
+            #       packCol = false ;
+            #    }
+            #    else {
+            #       msgTypeCol = baseType ;
+            #       packCol = true ;
+            #    }
+            # }
+            @assert !ALLOW_UNPACKED_COL
+            packCol = true
+
+            if colMin
+               # contiguous memory
+               if packCol
+                  srcOffset = 0
+                  offset = pmsg * maxPlaneComm
+                  for field in fields
+                     for i in 0:(dz-1)
+                        for j in 0:(dy-1)
+                           # INDEXING +1 missign somewhere
+                           domain.commDataSend[offset+i*dx+j] = field[srcOffset+i*dx*dy+j*dx]
```

Comment:
Field requires a +1 for 1-indexed, dest addr is i*DY not i*dx, https://github.com/wsmoses/Enzyme-MPI/blob/d211d3522628628cc3990a2470639b3159104a37/lulesh-comm.cc#L503